### ISSUE TITLE
Malformed x-b3-traceid Header

### ISSUE THREAD

**Issue by atze234** [containous/traefik#5842]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

When enabling tracing with b3 propagation and gen128Bit the x-b3-traceid is not 32 digits long. 

### What did you expect to see?

X-B3-Traceid: 4b03a264f5ae470241e7619143ef228

### What did you see instead?

X-B3-Traceid: 04b03a264f5ae470241e7619143ef228

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
/ # traefik version
Version:      2.0.5
Codename:     montdor
Go version:   go1.13.4
Built:        2019-11-14T18:11:01Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```
      --entryPoints.web.address=:8000
      --entryPoints.web.forwardedheaders.insecure=true
      --entryPoints.websecure.address=:8443
      --entryPoints.traefik.address=:9000
      --api.dashboard=true
      --api.insecure=true
      --accesslog
      --log
      --ping=true
      --providers.kubernetescrd
      --providers.kubernetescrd.labelselector=traffic-type=private
      --providers.kubernetesingress
      --providers.kubernetesingress.ingressclass=traefik2-private
      --log.level=DEBUG
      --tracing.jaeger=true
      --tracing.jaeger.samplingServerURL="http://jaeger-agent.tracing.svc.cluster.local:5778/sampling"
      --tracing.jaeger.localAgentHostPort=jaeger-agent.tracing.svc.cluster.local:6831
      --tracing.jaeger.samplingType=const
      --tracing.jaeger.samplingParam=1.0
      --tracing.jaeger.propagation=b3
      --tracing.jaeger.gen128Bit=true

```

I think its the same issue as in #4712

**Comment by ldez** [comment]
ping @jcchavezs

**Comment by ldez** [comment]
Hello,

It's an issue of https://github.com/jaegertracing/jaeger-client-go

**Comment by ldez** [comment]
@atze234 could you open an issue on https://github.com/jaegertracing/jaeger-client-go, thanks.

**Comment by jcchavezs** [comment]
Yeah that is what I was about to say because in zipkin-go we make sure the lenght of the trace_ids: https://github.com/openzipkin/zipkin-go/blob/master/model/traceid_test.go#L24 and https://github.com/openzipkin/zipkin-go/blob/master/model/traceid_test.go#L56

**Comment by ldez** [comment]
https://github.com/jaegertracing/jaeger/issues/1657
https://github.com/jaegertracing/jaeger-client-go/issues/471

**Comment by ldez** [comment]
@jcchavezs the zipkin package in jaeger-client-go seems also contains a bug:

https://github.com/jaegertracing/jaeger-client-go/blob/a015c340eda344ef534e88cef4f2604c211f5810/zipkin/propagation.go#L62-L66

**Comment by jcchavezs** [comment]
yeah that is probably true. I bet they will fix it now that you reported it.

**Comment by traefiker** [comment]
Closed by #6079.
